src/Orleans/Core/ClusterClient.cs

```diff
@@ -143,31 +164,36 @@ public static ClusterClient Create(FileInfo configFile)
         public static ClusterClient Create(ClientConfiguration configuration) => new ClusterClient(configuration);
 
         /// <inheritdoc />
-        public void Start()
+        public async Task Start()
         {
-            lock (this.initLock)
+            this.ThrowIfDisposed();
+            using (await this.initLock.LockAsync())
```

Comment:
I think the awaits in OutsideRuntimeClient are fine, since the public Start method does have ConfigureAwait(false) - what do you think?